audittools/rabbitmq.go

```diff
@@ -81,36 +79,43 @@ func NewRabbitConnection(uri, queueName string) (*RabbitConnection, error) {
 func (c *RabbitConnection) Disconnect() {
 	c.Channel.Close()
 	c.Inner.Close()
-	c.IsConnected = false
+}
+
+// IsNilOrClosed is like (*amqp.Connection).IsClosed() but it also returns true
+// if RabbitConnection or the underlying amqp.Connection are nil.
+func (c *RabbitConnection) IsNilOrClosed() bool {
+	if c != nil {
+		if c.Inner != nil && !c.Inner.IsClosed() {
```

Comment:
shall we check whether a `c.Channel` is closed as well? there could be a race condition, when a channel is closed, but a connection is still opened. Probably it's worth to add a lock to a RabbitConnection struct.

Reply:
I don't think there is any need for that. From [`func (*Connection) Close`](https://pkg.go.dev/github.com/streadway/amqp?utm_source=gopls#Connection.Close):

> An error indicates that server may not have received this request to close but the connection should be treated as closed regardless.
>
> After returning from this call, all resources associated with this connection, including the underlying io, Channels, Notify listeners and Channel consumers will also be closed. 

So after we call `func (*RabbitConnection) Disconnect`, we treat the underlying connection and the channel associated with it as closed/unusable and proceed with getting a new connection.

There is no need to add a lock to `RabbitConnection` because it is never used by multiple actors at the same time.